src/keep_gpu/cli.py

```
@@ -32,79 +93,172 @@ def _apply_legacy_threshold(
+def _http_json_request(
+    method: str,
+    url: str,
+    payload: Optional[Dict[str, Any]] = None,
+    timeout: float = 2.0,
+) -> Dict[str, Any]:
+    data = None
+    headers = {"content-type": "application/json"}
+    if payload is not None:
+        data = json.dumps(payload).encode("utf-8")
+    request = Request(url=url, data=data, headers=headers, method=method)
+    with urlopen(request, timeout=timeout) as response:  # nosec B310
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)